OrbitCore/LinuxEventTracerVisitor.cpp

```diff
@@ -0,0 +1,50 @@
+#include "OrbitProcess.h"
+#include "ContextSwitch.h"
+#include "TimerManager.h"
+#include "LinuxEventTracerVisitor.h"
+#include "LinuxPerfEvent.h"
+#include "PrintVar.h"
+
+void LinuxEventTracerVisitor::visit(LinuxPerfLostEvent* e)
+{
+    PRINT("Lost %u Events\n", e->Lost());
+}
+
+void LinuxEventTracerVisitor::visit(LinuxForkEvent* e)
+{
+    if (m_Process->HasThread(e->ParentTID()))
+    {
+        m_Process->AddThreadId(e->TID());
+    }
+}
+
+void LinuxEventTracerVisitor::visit(LinuxSchedSwitchEvent* e)
+{
+    // the known thread stopped running
+    if (m_Process->HasThread(e->PrevPID()))
+    {
+        ++Capture::GNumContextSwitches;
+
+        ContextSwitch CS ( ContextSwitch::Out );
+        CS.m_ThreadId = e->PrevPID();
+        CS.m_Time = e->Timestamp();
+        CS.m_ProcessorIndex = e->CPU();
+        //TODO: Is this correct?
+        CS.m_ProcessorNumber = e->CPU();
```

Comment:
I commented already elsewhere, but not sure if that comment is still visible: Is there a reason we have both ProcessorIndex and ProcessorNumber? Where is this actually different?

Reply:
Yes it is still visible, but as this is the current version, I copy over my response and close the one on the outdated commit:

ProcessorNumber is the number of CPUs in the machine, ProcessorIndex is the core index. Currently ProcessorNumber is not used by Orbit at all. In particular, I don't think that this information is available while tracing. We could use 0 for ProcessorNumber otherwise. In the end, it basically needs to be consistent.